src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java

```diff
@@ -399,6 +399,9 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
 							throwUnchecked(new InterruptedException());
 						}
 						awaitResult = stepSlice.await(1, TimeUnit.MILLISECONDS);
+						if (awaitResult) { // awaitResult = (0 == countDown)
```

Comment:
Could you please share some knowledge on how this works? Why awaitResult==true means that we've reached the limit for items? I was expecting that there is a necessity to wait instead of closing everything.

Reply:
awaitResult is the return value from the await() method. If you go into this method, you will see that this method returns (0 == countDown). countDown is a reversible counter for the count limit, and if it is equal to 0, this will mean that the count limit has been reached.